docs/content/overriding-styles.mdx

```diff
@@ -0,0 +1,59 @@
+---
+title: Overriding styles with the sx prop
+---
+
+Our goal with Primer Components is to hit the sweet spot between providing too little and too much styling flexibility; too little and the design system is too rigid, and too much and it becomes too difficult to maintain a consistent style. Our components already support a standard set of [system props](/system-props), but sometimes a component just isn't *quite* flexible enough to look the way you need it to look. For those cases, we provide the `sx` prop.
+
+The `sx` prop allows ad-hoc styling that is still theme aware. Declare the styles you want to apply in camel-cased object notation, and any values use appropriate theme values; for example, any `color` property values will use the values from the `colors` key in the theme.
```

Comment:
This above section might be a little confusing to folks who aren't totally familiar with how pulling from the theme works. You could link to those docs: https://primer.style/components/primer-theme

I think this paragraph also makes it seem like you can _only_ pull from the theme for `sx` styles which isn't true. Maybe we can change the wording a bit to communicate that we'd like people to pull from the theme whenever possible? Something like this? (still needs some word smithing):

```suggestion
The `sx` prop allows ad-hoc styling that is still theme aware. Declare the styles you want to apply in camel-cased object notation and try to use theme values whenever possible for rule values. For more information about how theming works, reference the [theme docs](https://primer.style/components/primer-theme).
```